pkg/wal/wal.go

```diff
@@ -0,0 +1,46 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package wal (Write-ahead logging) is an independent component to ensure data reliability.
+package wal
+
+// Options for Write-ahead Logging.
+type Options struct{}
+
+// Segment stands for a segment instance of Write-ahead log.
+type Segment interface{}
+
+// WAL includes exposed interfaces.
+type WAL interface {
+	// Write request to the WAL buffer.
+	// It will return synchronously when the request write in the WAL buffer,
+	// and trigger asynchronous callback when return when the buffer is flushed to disk successfully.
```

Comment:
```suggestion
	// Write a logging entity.
	// It will return immediately when the data is written in the buffer,
	// The returned function will be called when the entity is flushed on the persistent storage.
```